### DISCUSSION TITLE
Seek to audio

### DISCUSSION THREAD

**Discussion by Squigglez2** [zauberzeug/nicegui#1636] [Ideas / Feature Requests]
It would be nice to add methods to seek, and possibly play/pause ui.audio

**Comment by falkoschindler** [comment]
Sure, that would be a great addition. Anyone up for a pull request?

Here are workarounds using `run_javascript` (same for audio):

```py
v = ui.video('https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4')
ui.button('Play', on_click=lambda: ui.run_javascript(f'getElement({v.id}).$el.play()', respond=False))
ui.button('Pause', on_click=lambda: ui.run_javascript(f'getElement({v.id}).$el.pause()', respond=False))
ui.button('Seek(5)', on_click=lambda: ui.run_javascript(f'getElement({v.id}).$el.currentTime = 5', respond=False))
```

**Reply by wgong** [reply]
@falkoschindler I like to give it a try, will report back after code-analysis

**Reply by wgong** [reply]
@Squigglez2  the built-in control allows one to move to a specific timestamp manually,  can you clarify more on the seek behavior? do you like to have a user-entered seek position?  See screenshot below
![image](https://github.com/zauberzeug/nicegui/assets/329928/4dc0e405-ce9d-41e6-a4fc-02f24e9f7ee7)

@falkoschindler I tried your js workaround,  it worked as expected. Play and Pause buttons are not needed because the builtin play button does the toggle. Do you envision the enhanced `ui.audio` to add a seek position input? Which element should I review as an example for this kind of enhancement?

**Reply by falkoschindler** [reply]
@wgong I don't think the UI should change. We just need new methods to play, pause and seek.

**Reply by wgong** [reply]
@falkoschindler I see, the builtin control displays the media length, I like to validate the seek position to be within the max, is that max number available in audio.js?

**Reply by Squigglez2** [reply]
@wgong I probably should've mentioned it in the initial post but I'm trying to run with controls off, since I'm just playing short ~2s audio clips. I was hoping for a built in method since I'm not too familiar with the media players and without the workarounds the files just get stuck at the end and I can't use them again.

**Reply by falkoschindler** [reply]
@wgong I'm not sure if the length is available in JavaScript. Until the audio is completely loaded, the length is probably unknown.

**Reply by wgong** [reply]
Hi @falkoschindler @rodja  
a PR at https://github.com/zauberzeug/nicegui/pull/1672 is created for your review

I tested it with an example under examples/audio_video/main.py

Thanks,

**Comment by wgong** [comment]
@falkoschindler I have created a new PR following your suggestion - https://github.com/zauberzeug/nicegui/pull/1741
